tests/integration/test_charm.py

```diff
@@ -61,67 +64,66 @@ async def test_build_and_deploy(ops_test: OpsTest):
-        f"{APP_NAME}:{APP_GRAFANA_DASHBOARD_DEVICES}",
-        f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_GRAFANA_DASHBOARD}",
+        # Deploy grafana-agent
+        ops_test.model.deploy(GRAFANA_AGENT_APP, channel="latest/stable")
     )
 
-    logger.info(
-        "Adding relation: %s:%s and %s:%s",
-        APP_NAME,
-        APP_LOKI_ALERT_RULE_FILES_DEVICES,
-        GRAFANA_AGENT_APP,
-        GRAFANA_AGENT_LOGGING_PROVIDER,
-    )
-    await ops_test.model.integrate(
-        f"{APP_NAME}:{APP_LOKI_ALERT_RULE_FILES_DEVICES}",
-        f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_LOGGING_PROVIDER}",
+    # and wait for active/idle status
+    await asyncio.gather(
+        ops_test.model.wait_for_idle(
+            apps=[APP_NAME, PROMETHEUS_APP], status="active", timeout=600
+        ),
+        ops_test.model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=600)
     )
 
-    logger.info(
-        "Adding relation: %s:%s and %s:%s",
-        APP_NAME,
-        APP_PROMETHEUS_ALERT_RULE_FILES_DEVICES,
-        PROMETHEUS_APP,
-        PROMETHEUS_RECEIVE_REMOTE_WRITE,
-    )
-    await ops_test.model.integrate(
-        f"{APP_NAME}:{APP_PROMETHEUS_ALERT_RULE_FILES_DEVICES}",
-        f"{PROMETHEUS_APP}:{PROMETHEUS_RECEIVE_REMOTE_WRITE}",
-    )
 
-    logger.info(
-        "Adding relation: %s:%s and %s:%s",
-        APP_NAME,
-        "tracing",
-        GRAFANA_AGENT_APP,
-        "tracing-provider",
+@pytest.mark.abort_on_fail
+async def test_integrate(ops_test: OpsTest):
+    await asyncio.gather(
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_GRAFANA_DASHBOARD}",
+            f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_GRAFANA_DASHBOARD}",
+        ),
+
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_LOGGING}",
+            f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_LOGGING_PROVIDER}",
+        ),
+
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_GRAFANA_DASHBOARD_DEVICES}",
+            f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_GRAFANA_DASHBOARD}",
+        ),
+
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_LOKI_ALERT_RULE_FILES_DEVICES}",
+            f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_LOGGING_PROVIDER}",
+        ),
+
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_PROMETHEUS_ALERT_RULE_FILES_DEVICES}",
+            f"{PROMETHEUS_APP}:{PROMETHEUS_RECEIVE_REMOTE_WRITE}",
+        ),
+
+        ops_test.model.integrate(
+            f"{APP_NAME}:{APP_TRACING}",
+            f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_TRACING_PROVIDER}",
+        )
     )
-    await ops_test.model.integrate(
-        f"{APP_NAME}:tracing",
-        f"{GRAFANA_AGENT_APP}:tracing-provider",
+
+    await asyncio.gather(
+        ops_test.model.wait_for_idle(
+            apps=[APP_NAME, PROMETHEUS_APP], status="active", raise_on_blocked=True, timeout=180
+        ),
+        ops_test.model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=180)
```

Comment:
Why reducing so much the timeout? I am really not sure this is going to work on my slow VM.

Reply:
3mins solely for waiting on the app integrations, that's too little? I don;t mind increasing it...